packages/core/src/components/EmptyState/EmptyState.types.ts

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { LinkProps } from "../Link/Link";
+import { VibeComponentProps } from "src/types";
+import { ButtonProps, ButtonType } from "../Button";
+
+export type EmptyStateLayout = "default" | "compact";
+
+export interface EmptyStateProps extends VibeComponentProps {
+  /** Optional title for the empty state */
+  title?: string;
+  /** Required description text explaining the empty state */
+  description: string | React.ReactNode;
+  /** Optional visual element like image, animation, video, or illustration to display */
+  visual?: React.ReactNode;
+  /** Main action button configuration */
+  mainAction?:
+    | React.ReactElement<ButtonProps>
+    | (Omit<ButtonProps, "kind" | "children" | "size"> & {
+        kind?: Exclude<ButtonType, "tertiary">;
+        text: string;
+      });
```

Comment:
It can be a local type here in this file just to make it clearer but not critical